cura/Settings/PerObjectContainerStack.py

```diff
@@ -17,6 +18,13 @@ def getProperty(self, key: str, property_name: str, context: Optional[PropertyEv
         from cura.CuraApplication import CuraApplication
         global_stack = CuraApplication.getInstance().getGlobalContainerStack()
 
+        # Return the user defined value if present, otherwise, evaluate the value according to the default routine.
+        if self.getContainer(0).hasProperty(key, property_name):
+            if self.getContainer(0)._instances[key].state == InstanceState.User:
+                result = super().getProperty(key, property_name, context)
+                context.popContainer()
+                return result
```

Comment:
User defined values are never formulas, so I don't see why this is necessary or why this is specific to per-object settings.